docs/pages/components/Button.mdx

```diff
@@ -36,12 +36,18 @@ Tertiary buttons are an alternate style of button with less visual prominence th
 <Button variant="tertiary">Tertiary</Button>
 ```
 
-### Error
+### Danger
```

Comment:
Note: This is not removing error, but removing it from documentation so that its not used going forward. We don't have a good way to deprecate props, but removing from documentation should help to show it's not being used and we can communicate out to other teams that "Error" is now deprecated.

Tracked by https://github.com/dequelabs/cauldron/issues/2075